OpenDreamShared/Compiler/DM/DMParser.cs

```diff
@@ -913,19 +913,21 @@ public class DMParser : Parser<Token> {
         }
 
         public DMASTDefinitionParameter[] DefinitionParameters() {
-            List<DMASTDefinitionParameter> parameters = new List<DMASTDefinitionParameter>();
+            List<DMASTDefinitionParameter> parameters = new();
             DMASTDefinitionParameter parameter = DefinitionParameter();
 
-            while (parameter != null) {
+            if (parameter != null || Check(TokenType.DM_IndeterminateArgs)) {
                 parameters.Add(parameter);
```

Comment:
I just realized this adds null to parameters if the first argument is DM_IndeterminateArgs
```suggestion
            if (parameter != null || Check(TokenType.DM_IndeterminateArgs)) {
                if (parameter != null) parameters.Add(parameter);
```